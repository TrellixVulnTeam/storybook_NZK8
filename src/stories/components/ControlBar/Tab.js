// @flow

import React from 'react';
import { Tab } from './styled';

import type { ControlBarItem } from './index';

type Props = {
  item: ControlBarItem,
  onSelectItem: ControlBarItem => void,
  theme?: string,
};

export default (props: Props) => {
  const active = Boolean(props.item.value);

  return (
    <Tab active={active} onClick={() => props.onSelectItem(props.item)} theme={props.theme}>
      {props.item.label}
    </Tab>
  );
};
