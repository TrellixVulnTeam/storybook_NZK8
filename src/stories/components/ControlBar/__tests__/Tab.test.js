import React from 'react';
import { mount } from 'enzyme';
import Tab from 'components/ControlBar/Tab';

describe('<Tab />', () => {
  const onSelectItemMock = jest.fn();
  const mandatoryProps = {
    onSelectItem: onSelectItemMock,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with full props', () => {
    const props = {
      ...mandatoryProps,
      item: {
        label: 'label',
        value: true,
      },
    };
    const sut = mount(<Tab {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly if not active', () => {
    const props = {
      ...mandatoryProps,
      item: {
        label: 'label',
        value: false,
      },
    };
    const sut = mount(<Tab {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should select item', () => {
    const props = {
      ...mandatoryProps,
      item: {
        label: 'label',
        value: true,
      },
    };
    const sut = mount(<Tab {...props} />);

    expect(onSelectItemMock).toHaveBeenCalledTimes(0);

    sut.find(Tab).simulate('click');

    expect(onSelectItemMock).toHaveBeenCalledTimes(1);
  });
});
