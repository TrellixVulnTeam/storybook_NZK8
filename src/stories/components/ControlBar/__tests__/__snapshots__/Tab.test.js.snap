// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> should render correctly if not active 1`] = `
<_default
  item={
    Object {
      "label": "label",
      "value": false,
    }
  }
  onSelectItem={[MockFunction]}
>
  <styled__Tab
    active={false}
    onClick={[Function]}
  >
    <div
      className="styled__Tab-xs9hpj-2 aVytJ"
      onClick={[Function]}
    >
      label
    </div>
  </styled__Tab>
</_default>
`;

exports[`<Tab /> should render correctly with full props 1`] = `
<_default
  item={
    Object {
      "label": "label",
      "value": true,
    }
  }
  onSelectItem={[MockFunction]}
>
  <styled__Tab
    active={true}
    onClick={[Function]}
  >
    <div
      className="styled__Tab-xs9hpj-2 gUllGI"
      onClick={[Function]}
    >
      label
    </div>
  </styled__Tab>
</_default>
`;
